services/api-server/src/simcore_service_api_server/models/schemas/solvers.py

```diff
@@ -0,0 +1,170 @@
+import re
+from datetime import datetime
+from enum import Enum
+from typing import List, Optional, Union
+from uuid import UUID
+
+from pydantic import BaseModel, Field, HttpUrl, conint, constr
+
+LATEST_VERSION = "latest"
+KEY_RE = r"^(simcore)/(services)/(comp)(/[^\s/]+)+$"  # NOTE: needs to end with / !!
+
+
+# Human-readable unique identifier
+KeyIdentifier = constr(strip_whitespace=True, min_length=3)
+SolverImageName = constr(regex=KEY_RE, strip_whitespace=True)
+
+
+class SolverImage(BaseModel):
+    # This is an image. Notice that tags refer to this image
+    uuid: UUID = Field(..., description="Image sha256 unique identifier")
+    name: SolverImageName = Field(
+        ...,
+        description="Name of the solver image including namespace and excluding tag",
+    )
+    maintainer: str
+    released: datetime
+
+
+class Solver(BaseModel):
+    """A released solver with a specific version
+
+    This version might have human-readable alias (e.g. latest) or
+    hierarchical version tags (e.g. 3, 3.2)
+    """
+
+    uuid: UUID = Field(..., description="Same as the solver's image sha256")
+    name: str = Field(..., description="Image name including namespace")
+    version: str  # complete tag.  e.g. 3.4.5 TODO: regex for version in python PEP
+    version_aliases: List[str] = []  # remaining tags
```

Comment:
does it make sense that this has absolutely nothing to do with the ServiceDockerData in models_library? this looks a bit like code duplication in some parts. Could we at least have some common ground?

Reply:
Good point. The PR is so old that both developed in parallel and I did not notice. thx